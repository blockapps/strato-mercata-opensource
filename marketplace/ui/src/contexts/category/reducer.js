import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createCategory:
      return {
        ...state,
        isCreateCategorySubmitting: true,
      };
    case actionDescriptors.createCategorySuccessful:
      return {
        ...state,
        category: action.payload,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.createCategoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateCategorySubmitting: false,
      };
    case actionDescriptors.fetchCategory:
      return {
        ...state,
        iscategorysLoading: true,
      };
    case actionDescriptors.fetchCategorySuccessful:
      return {
        ...state,
        categorys: action.payload,
        iscategorysLoading: false,
      };
    case actionDescriptors.fetchCategoryFailed:
      return {
        ...state,
        error: action.error,
        iscategorysLoading: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
