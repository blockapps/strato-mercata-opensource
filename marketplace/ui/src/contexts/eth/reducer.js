import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.fetchETHSTAddress:
      return {
        ...state,
      };
    case actionDescriptors.fetchETHSTAddressSuccessful:
      return {
        ...state,
        ethstAddress: action.payload,
      };
    case actionDescriptors.fetchETHSTAddressFailed:
      return {
        ...state,
      };
    case actionDescriptors.addHash:
      return {
        ...state,
        isAddingHash: true,
      };
    case actionDescriptors.addHashSuccessful:
      return {
        ...state,
        isAddingHash: false,
      };
    case actionDescriptors.addHashFailed:
      return {
        ...state,
        isAddingHash: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
