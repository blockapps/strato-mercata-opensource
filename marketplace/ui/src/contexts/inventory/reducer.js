import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.createInventory:
      return {
        ...state,
        isCreateInventorySubmitting: true,
      };
    case actionDescriptors.createInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.createInventoryFailed:
      return {
        ...state,
        error: action.error,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.fetchInventory:
      return {
        ...state,
        isInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventorySuccessful:
      return {
        ...state,
        inventories: action.payload.data,
        inventoriesTotal: action.payload.count,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryFailed:
      return {
        ...state,
        error: action.error,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryForUser:
      return {
        ...state,
        isUserInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventoryForUserSuccessful:
      return {
        ...state,
        userInventories: action.payload.data,
        userInventoriesTotal: action.payload.count,
        isUserInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventoryForUserFailed:
      return {
        ...state,
        error: action.error,
        isUserInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventorySearch:
      return {
        ...state,
        isInventoriesLoading: true,
      };
    case actionDescriptors.fetchInventorySearchSuccessful:
      return {
        ...state,
        inventories: action.payload.data,
        inventoriesTotal: action.payload.count,
        isInventoriesLoading: false,
      };
    case actionDescriptors.fetchInventorySearchFailed:
      return {
        ...state,
        error: action.error,
        isInventoriesLoading: false,
      };
    case actionDescriptors.updateInventory:
      return {
        ...state,
        isinventoryUpdating: true,
      };
    case actionDescriptors.updateInventorySuccessful:
      return {
        ...state,
        inventoryUpdateObject: action.payload,
        isinventoryUpdating: false,
      };
    case actionDescriptors.updateInventoryFailed:
      return {
        ...state,
        error: action.error,
        isinventoryUpdating: false,
      };
    case actionDescriptors.updateSale:
      return {
        ...state,
        issaleUpdating: true,
      };
    case actionDescriptors.updateSaleSuccessful:
      return {
        ...state,
        saleUpdateObject: action.payload,
        issaleUpdating: false,
      };
    case actionDescriptors.updateSaleFailed:
      return {
        ...state,
        error: action.error,
        issaleUpdating: false,
      };
    case actionDescriptors.listInventory:
      return {
        ...state,
        isListing: true,
      };
    case actionDescriptors.listInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isListing: false,
      };
    case actionDescriptors.listInventoryFailed:
      return {
        ...state,
        error: action.error,
        isListing: false,
      };
    case actionDescriptors.unlistInventory:
      return {
        ...state,
        isUnlisting: true,
      };
    case actionDescriptors.unlistInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isUnlisting: false,
      };
    case actionDescriptors.unlistInventoryFailed:
      return {
        ...state,
        error: action.error,
        isUnlisting: false,
      };

     // ------------------------------------------------------------------------------------------------------
    case actionDescriptors.stakeInventory:
      return {
        ...state,
        isStaking: true
      };
    case actionDescriptors.stakeInventorySuccessful:
      return {
        ...state,
        isStaking: false
      };
    case actionDescriptors.stakeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isStaking: false
      };
    
    case actionDescriptors.unstakeInventory:
      return {
        ...state,
        isUnstaking: true
      };
    case actionDescriptors.unstakeInventorySuccessful:
      return {
        ...state,
        isUnstaking: false
      };
    case actionDescriptors.unstakeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isUnstaking: false
      };
    
    case actionDescriptors.borrow:
      return {
        ...state,
        isBorrowing: true
      };
    case actionDescriptors.borrowSuccessful:
      return {
        ...state,
        isBorrowing: false
      };
    case actionDescriptors.borrowFailed:
      return {
        ...state,
        error: action.error,
        isBorrowing: false
      };
    
    case actionDescriptors.repay:
      return {
        ...state,
        isRepaying: true
      };
    case actionDescriptors.repaySuccessful:
      return {
        ...state,
        isRepaying: false
      };
    case actionDescriptors.repayFailed:
        return {
          ...state,
          error: action.error,
          isRepaying: false
        };

    case actionDescriptors.getReserve:
        return {
          ...state,
          isReserveLoading: true
        };
    case actionDescriptors.getReserveSuccessful:
        return {
          ...state,
          reserve: action.payload,
          isReserveLoading: false
        };
    case actionDescriptors.getReserveFailed:
        return {
          ...state,
          error: action.error,
          isReserveLoading: false
        };
    
    case actionDescriptors.getAllReserve:
        return {
          ...state,
          isReservesLoading: true
        };
    case actionDescriptors.getAllReserveSuccessful:
        return {
          ...state,
          reserves: action.payload,
          isReservesLoading: false
        };
    case actionDescriptors.getAllReserveFailed:
        return {
          ...state,
          error: action.error,
          isReservesLoading: false
        };
    
    case actionDescriptors.getEscrowForAsset:
        return {
          ...state,
          isEscrowLoading: true
        };
    case actionDescriptors.getEscrowForAssetSuccessful:
        return {
          ...state,
          escrow: action.payload,
          isEscrowLoading: false
        };
    case actionDescriptors.getEscrowForAssetFailed:
        return {
          ...state,
          error: action.error,
          isEscrowLoading: false
        };

    case actionDescriptors.getUserCataRewards:
        return {
          ...state,
          isUserCataRewardsLoading: true
        };
    case actionDescriptors.getUserCataRewardsSuccessful:
        return {
          ...state,
          totalCataReward: action.payload.totalCataReward,
          dailyCataReward: action.payload.dailyCataReward,
          isUserCataRewardsLoading: false
        };
    case actionDescriptors.getUserCataRewardsFailed:
        return {
          ...state,
          error: action.error,
          isUserCataRewardsLoading: false
        };
    
    case actionDescriptors.getOracle:
          return {
            ...state,
            isFetchingOracle: true
          };
    case actionDescriptors.getOracleSuccessful:
          return {
            ...state,
            oracle: action.payload,
            isFetchingOracle: false
          };
    case actionDescriptors.getOracleFailed:
          return {
            ...state,
            error: action.error,
            isFetchingOracle: false
          };    
    // ------------------------------------------------------------------------------------------------------
    case actionDescriptors.resellInventory:
      return {
        ...state,
        isReselling: true,
      };
    case actionDescriptors.resellInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isReselling: false,
      };
    case actionDescriptors.resellInventoryFailed:
      return {
        ...state,
        error: action.error,
        isReselling: false,
      };
    case actionDescriptors.fetchSupportedTokens:
      return {
        ...state,
        isFetchingTokens: true,
      };
    case actionDescriptors.fetchSupportedTokensSuccessful:
      return {
        ...state,
        supportedTokens: action.payload,
        isFetchingTokens: false,
      };
    case actionDescriptors.fetchSupportedTokensFailed:
      return {
        ...state,
        error: action.error,
        isFetchingTokens: false,
      };
    case actionDescriptors.bridgeInventory:
      return {
        ...state,
        isBridging: true,
      };
    case actionDescriptors.bridgeInventorySuccessful:
      return {
        ...state,
        inventory: action.payload,
        isBridging: false,
      };
    case actionDescriptors.bridgeInventoryFailed:
      return {
        ...state,
        error: action.error,
        isBridging: false,
      };
    case actionDescriptors.transferInventory:
      return {
        ...state,
        isTransferring: true,
      };
    case actionDescriptors.transferInventorySuccessful:
      return {
        ...state,
        isTransferring: false,
      };
    case actionDescriptors.transferInventoryFailed:
      return {
        ...state,
        error: action.error,
        isTransferring: false,
      };
    case actionDescriptors.fetchItemTransfers:
      return {
        ...state,
        isFetchingItemTransfers: true,
      };
    case actionDescriptors.fetchItemTransfersSuccessful:
      return {
        ...state,
        itemTransfers: action.payload.transfers,
        totalItemsTransfered: action.payload.total,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemTransfersFailed:
      return {
        ...state,
        error: action.error,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchInventoryOwnershipHistory:
      return {
        ...state,
        isInventoryOwnershipHistoryLoading: true,
      };
    case actionDescriptors.fetchInventoryOwnershipHistorySuccessful:
      return {
        ...state,
        inventoryOwnershipHistory: action.payload,
        isInventoryOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchInventoryOwnershipHistoryFailed:
      return {
        ...state,
        error: action.error,
        isInventoryOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchInventoryDetail:
      return {
        ...state,
        isInventoryDetailsLoading: true,
      };
    case actionDescriptors.fetchInventoryDetailSuccessful:
      return {
        ...state,
        inventoryDetails: action.payload,
        isInventoryDetailsLoading: false,
      };
    case actionDescriptors.fetchInventoryDetailFailed:
      return {
        ...state,
        error: action.error,
        isInventoryDetailsLoading: false,
      };
    case actionDescriptors.uploadImage:
      return {
        ...state,
        isUploadImageSubmitting: true,
      };
    case actionDescriptors.uploadImageSuccessful:
      return {
        ...state,
        uploadedImg: action.payload,
        isUploadImageSubmitting: false,
      };
    case actionDescriptors.uploadImageFailed:
      return {
        ...state,
        error: action.error,
        isUploadImageSubmitting: false,
      };
    case actionDescriptors.createItem:
      return {
        ...state,
        isCreateInventorySubmitting: true,
      };
    case actionDescriptors.createItemSuccessful:
      return {
        ...state,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.createItemFailed:
      return {
        ...state,
        error: action.error,
        isCreateInventorySubmitting: false,
      };
    case actionDescriptors.fetchPriceHistory:
      return {
        ...state,
        isFetchingPriceHistory: true,
      };
    case actionDescriptors.fetchPriceHistorySuccessful:
      return {
        ...state,
        isFetchingPriceHistory: false,
        priceHistory: action.payload,
      };
    case actionDescriptors.fetchPriceHistoryFailed:
      return {
        ...state,
        isFetchingPriceHistory: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
