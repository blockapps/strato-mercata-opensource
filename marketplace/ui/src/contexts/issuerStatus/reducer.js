import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.requestReview:
      return {
        ...state,
        requestingReview: true,
      };
    case actionDescriptors.requestReviewSuccessful:
      return {
        ...state,
        requestingReview: false,
      };
    case actionDescriptors.requestReviewFailed:
      return {
        ...state,
        requestingReview: false,
      };
    case actionDescriptors.authorizeIssuer:
      return {
        ...state,
        changingIssuerStatus: true,
      };
    case actionDescriptors.authorizeIssuerSuccessful:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.authorizeIssuerFailed:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.deauthorizeIssuer:
      return {
        ...state,
        changingIssuerStatus: true,
      };
    case actionDescriptors.deauthorizeIssuerSuccessful:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.deauthorizeIssuerFailed:
      return {
        ...state,
        changingIssuerStatus: false,
      };
    case actionDescriptors.modifyAdmin:
      return {
        ...state,
        changingAdminStatus: true,
      };
    case actionDescriptors.modifyAdminSuccessful:
      return {
        ...state,
        changingAdminStatus: false,
      };
    case actionDescriptors.modifyAdminFailed:
      return {
        ...state,
        changingAdminStatus: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
