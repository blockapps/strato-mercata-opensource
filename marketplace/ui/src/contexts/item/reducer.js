import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.fetchItem:
      return {
        ...state,
        isItemsLoading: true,
      };
    case actionDescriptors.fetchItemSuccessful:
      return {
        ...state,
        items: action.payload,
        isItemsLoading: false,
      };
    case actionDescriptors.fetchItemFailed:
      return {
        ...state,
        error: action.error,
        isItemsLoading: false,
      };
    case actionDescriptors.fetchSerialNumbers:
      return {
        ...state,
        isSerialNumbersLoading: true,
      };
    case actionDescriptors.fetchSerialNumbersSuccessful:
      return {
        ...state,
        serialNumbers: action.payload,
        isSerialNumbersLoading: false,
      };
    case actionDescriptors.fetchSerialNumbersFailed:
      return {
        ...state,
        error: action.error,
        isSerialNumbersLoading: false,
      };
    case actionDescriptors.fetchItemOwnershipHistory:
      return {
        ...state,
        isOwnershipHistoryLoading: true,
      };
    case actionDescriptors.fetchItemOwnershipHistorySuccessful:
      return {
        ...state,
        ownershipHistory: action.payload,
        isOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchItemOwnershipHistoryFailed:
      return {
        ...state,
        error: action.error,
        isOwnershipHistoryLoading: false,
      };
    case actionDescriptors.fetchItemTransfers:
      return {
        ...state,
        isFetchingItemTransfers: true,
      };
    case actionDescriptors.fetchItemTransfersSuccessful:
      return {
        ...state,
        itemTransfers: action.payload.transfers,
        totalItemsTransfered: action.payload.total,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemTransfersFailed:
      return {
        ...state,
        error: action.error,
        isFetchingItemTransfers: false,
      };
    case actionDescriptors.fetchItemRawMaterials:
      return {
        ...state,
        isRawMaterialsLoading: true,
      };
    case actionDescriptors.fetchItemRawMaterialsSuccessful:
      return {
        ...state,
        rawMaterials: action.payload,
        isRawMaterialsLoading: false,
      };
    case actionDescriptors.fetchItemRawMaterialsFailed:
      return {
        ...state,
        error: action.error,
        isRawMaterialsLoading: false,
      };
    case actionDescriptors.setActualRawMaterials:
      return {
        ...state,
        actualRawMaterials: action.payload,
      };
    case actionDescriptors.transferOwnership:
      return {
        ...state,
        isTransferringItems: true,
      };
    case actionDescriptors.transferOwnershipSuccessful:
      return {
        ...state,
        isTransferringItems: false,
      };
    case actionDescriptors.transferOwnershipFailed:
      return {
        ...state,
        error: action.error,
        isTransferringItems: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
