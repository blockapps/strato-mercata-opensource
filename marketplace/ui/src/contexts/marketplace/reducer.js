import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.resetMessage:
      return {
        ...state,
        success: false,
        message: null,
      };
    case actionDescriptors.setMessage:
      return {
        ...state,
        success: action.success,
        message: action.message,
      };
    case actionDescriptors.fetchMarketplace:
      return {
        ...state,
        isMarketplaceLoading: true,
      };
    case actionDescriptors.fetchMarketplaceSuccessful:
      return {
        ...state,
        marketplaceList: action.payload.productsWithImageUrl,
        marketplaceListCount: action.payload.inventoryCount,
        isMarketplaceLoading: false,
      };
    case actionDescriptors.fetchMarketplaceFailed:
      return {
        ...state,
        error: action.error,
        isMarketplaceLoading: false,
      };
    case actionDescriptors.fetchMarketplaceLoggedIn:
      return {
        ...state,
        isMarketplaceLoading: true,
      };
    case actionDescriptors.fetchMarketplaceLoggedInSuccessful:
      return {
        ...state,
        marketplaceList: action.payload.productsWithImageUrl,
        marketplaceListCount: action.payload.inventoryCount,
        isMarketplaceLoading: false,
      };
    case actionDescriptors.fetchMarketplaceLoggedInFailed:
      return {
        ...state,
        error: action.error,
        isMarketplaceLoading: false,
      };
    case actionDescriptors.fetchCartItems:
      return {
        ...state,
      };
    case actionDescriptors.fetchCartItemsSuccessful:
      return {
        ...state,
        cartList: action.payload,
      };
    case actionDescriptors.fetchCartItemsFailed:
      return {
        ...state,
        error: action.error,
      };
    case actionDescriptors.addItemToCart:
      return {
        ...state,
      };
    case actionDescriptors.addItemToCartSuccessful:
      return {
        ...state,
        cartList: action.payload,
      };
    case actionDescriptors.addItemToCartFailed:
      return {
        ...state,
        error: action.error,
      };
    case actionDescriptors.fetchConfirmOrderItems:
      return {
        ...state,
      };
    case actionDescriptors.fetchConfirmOrderItemsSuccessful:
      return {
        ...state,
        confirmOrderList: action.payload,
      };
    case actionDescriptors.fetchConfirmOrderItemsFailed:
      return {
        ...state,
        error: action.error,
      };
    case actionDescriptors.addItemToConfirmOrder:
      return {
        ...state,
      };
    case actionDescriptors.addItemToConfirmOrderSuccessful:
      return {
        ...state,
        confirmOrderList: action.payload,
      };
    case actionDescriptors.addItemToConfirmOrderFailed:
      return {
        ...state,
        error: action.error,
      };
    case actionDescriptors.deleteCartItem:
      return {
        ...state,
      };
    case actionDescriptors.deleteCartItemSuccesful:
      return {
        ...state,
        cartList: action.payload,
      };
    case actionDescriptors.deleteCartItemFailed:
      return {
        ...state,
        error: action.error,
      };
    //top selling products
    case actionDescriptors.fetchTopSellingProducts:
      return {
        ...state,
        isTopSellingProductsLoading: true,
      };
    case actionDescriptors.fetchTopSellingProductsSuccessful:
      return {
        ...state,
        topSellingProducts: action.payload,
        isTopSellingProductsLoading: false,
      };
    case actionDescriptors.fetchTopSellingProductsFailed:
      return {
        ...state,
        error: action.error,
        isTopSellingProductsLoading: false,
      };
    case actionDescriptors.fetchTopSellingProductsLoggedIn:
      return {
        ...state,
        isTopSellingProductsLoading: true,
      };
    case actionDescriptors.fetchTopSellingProductsLoggedInSuccessful:
      return {
        ...state,
        topSellingProducts: action.payload,
        isTopSellingProductsLoading: false,
      };
    case actionDescriptors.fetchTopSellingProductsLoggedInFailed:
      return {
        ...state,
        error: action.error,
        isTopSellingProductsLoading: false,
      };

    case actionDescriptors.fetchStakeableProducts:
      return {
        ...state,
        isStakeableProductsLoading: true,
      };
    case actionDescriptors.fetchStakeableProductsSuccessful:
      return {
        ...state,
        stakeableProducts: action.payload,
        isStakeableProductsLoading: false,
      };
    case actionDescriptors.fetchStakeableProductsFailed:
      return {
        ...state,
        error: action.error,
        isStakeableProductsLoading: false,
      };

    //shipping address adding
    case actionDescriptors.addShippingAddress:
      return {
        ...state,
        isAddingShippingAddress: true,
      };
    case actionDescriptors.addShippingAddressSuccessful:
      return {
        ...state,
        shippingAddress: action.payload,
        isAddingShippingAddress: false,
      };
    case actionDescriptors.addShippingAddressFailed:
      return {
        ...state,
        error: action.error,
        isAddingShippingAddress: false,
      };
    case actionDescriptors.fetchUserAddress:
      return {
        ...state,
        isLoadingUserAddress: true,
      };
    case actionDescriptors.fetchUserAddressSuccessful:
      return {
        ...state,
        userAddress: action.payload,
        isLoadingUserAddress: false,
      };
    case actionDescriptors.fetchUserAddressFailed:
      return {
        ...state,
        error: action.error,
        isLoadingUserAddress: false,
      };
    case actionDescriptors.fetchUserAddresses:
      return {
        ...state,
        isLoadingUserAddresses: true,
      };
    case actionDescriptors.fetchUserAddressesSuccessful:
      return {
        ...state,
        userAddresses: action.payload,
        isLoadingUserAddresses: false,
      };
    case actionDescriptors.fetchUserAddressesFailed:
      return {
        ...state,
        error: action.error,
        isLoadingUserAddresses: false,
      };
    case actionDescriptors.fetchUSDSTBalance:
      return {
        ...state,
        isFetchingUSDST: true,
      };
    case actionDescriptors.fetchUSDSTBalanceSuccessful:
      return {
        ...state,
        isFetchingUSDST: false,
        USDST: action.payload,
      };
    case actionDescriptors.fetchUSDSTBalanceFailed:
      return {
        ...state,
        isFetchingUSDST: false,
      };
    case actionDescriptors.fetchCataBalance:
      return {
        ...state,
        isFetchingCata: true,
      };
    case actionDescriptors.fetchCataBalanceSuccessful:
      return {
        ...state,
        isFetchingCata: false,
        cata: action.payload,
      };
    case actionDescriptors.fetchCataBalanceFailed:
      return {
        ...state,
        isFetchingCata: false,
      };
    case actionDescriptors.fetchUSDSTAddress:
      return {
        ...state,
      };
    case actionDescriptors.fetchUSDSTAddressSuccessful:
      return {
        ...state,
        USDSTAddress: action.payload,
      };
    case actionDescriptors.fetchUSDSTAddressFailed:
      return {
        ...state,
      };
    case actionDescriptors.fetchStratsAddress:
      return {
        ...state,
      };
    case actionDescriptors.fetchStratsAddressSuccessful:
      return {
        ...state,
        stratsAddress: action.payload,
      };
    case actionDescriptors.fetchStratsAddressFailed:
      return {
        ...state,
      };
    case actionDescriptors.fetchAssetsWithEighteenDecimalPlaces:
      return {
        ...state,
        isFetchingAssetsWithEighteenDecimalPlaces: true,
      };
    case actionDescriptors.fetchAssetsWithEighteenDecimalPlacesSuccessful:
      return {
        ...state,
        isFetchingAssetsWithEighteenDecimalPlaces: false,
        assetsWithEighteenDecimalPlaces: action.payload,
      };
    case actionDescriptors.fetchAssetsWithEighteenDecimalPlacesFailed:
      return {
        ...state,
        isFetchingAssetsWithEighteenDecimalPlaces: false,
      };

    case actionDescriptors.fetchCataAddress:
      return {
        ...state,
      };
    case actionDescriptors.fetchCataAddressSuccessful:
      return {
        ...state,
        cataAddress: action.payload,
      };
    case actionDescriptors.fetchCataAddressFailed:
      return {
        ...state,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
