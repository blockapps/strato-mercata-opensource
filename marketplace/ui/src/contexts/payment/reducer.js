import { actionDescriptors } from './actions';

const reducer = (state, action) => {
  switch (action.type) {
    case actionDescriptors.getPaymentServices:
      return {
        ...state,
        arePaymentServicesLoading: true,
      };
    case actionDescriptors.getPaymentServicesSuccessful:
      return {
        ...state,
        paymentServices: action.payload.data,
        paymentServicesTotal: action.payload.count,
        arePaymentServicesLoading: false,
      };
    case actionDescriptors.getPaymentServicesFailed:
      return {
        ...state,
        error: action.error,
        arePaymentServicesLoading: false,
      };
    case actionDescriptors.getNotOnboarded:
      return {
        ...state,
        areNotOnboardedLoading: true,
      };
    case actionDescriptors.getNotOnboardedSuccessful:
      return {
        ...state,
        notOnboarded: action.payload.data,
        notOnboardedTotal: action.payload.count,
        areNotOnboardedLoading: false,
      };
    case actionDescriptors.getNotOnboardedFailed:
      return {
        ...state,
        error: action.error,
        areNotOnboardedLoading: false,
      };
    default:
      throw new Error(`Unhandled action: '${action.type}'`);
  }
};

export default reducer;
