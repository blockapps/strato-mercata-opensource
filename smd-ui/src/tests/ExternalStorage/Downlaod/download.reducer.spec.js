import reducer from "../../../components/ExternalStorage/Download/download.reducer";
import {
  openDownloadModal, closeDownloadModal, downloadSuccess, downloadFailure, resetError, clearUrl
} from "../../../components/ExternalStorage/Download/download.actions";
import { error } from "../storageMock";

describe('Download: reducer', () => {

  // INITIAL_STATE
  test('set initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  describe('Modal', () => {

    // OPEN_DOWNLOAD_MODAL
    test('open', () => {

      const initialState = {
        isOpen: false,
        error: null,
        url: null
      }

      const action = openDownloadModal();
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // CLOSE_DOWNLOAD_MODAL
    test('close', () => {

      const initialState = {
        isOpen: true,
        error: null,
        url: null
      }

      const action = closeDownloadModal();
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

  });

  describe('create user', () => {

    const initialState = {
      isOpen: false,
      error: null,
      url: null
    }

    // DOWNLOAD_SUCCESS
    test('on success', () => {
      const url = 'https://strato-external-storage.s3.amazonaws.com/1529915329415-widescreen.jpeg';
      const action = downloadSuccess(url);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

    // DOWNLOAD_FAILURE
    test('on failure', () => {
      const action = downloadFailure(error);
      expect(reducer(initialState, action)).toMatchSnapshot();
    });

  });

  // RESET_ERROR
  test('reset Error', () => {

    const initialState = {
      isOpen: false,
      error: 'error',
      url: null
    }

    const action = resetError();

    expect(reducer(initialState, action)).toMatchSnapshot();
  });

  // CLEAR_URL
  test('clear URL', () => {

    const initialState = {
      isOpen: false,
      error: 'error',
      url: null
    }

    const action = clearUrl();

    expect(reducer(initialState, action)).toMatchSnapshot();
  });

});