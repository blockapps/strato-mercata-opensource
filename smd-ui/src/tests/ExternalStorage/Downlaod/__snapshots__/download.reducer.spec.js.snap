// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Download: reducer Modal close 1`] = `
Object {
  "error": null,
  "isOpen": false,
  "url": null,
}
`;

exports[`Download: reducer Modal open 1`] = `
Object {
  "error": null,
  "isOpen": true,
  "url": null,
}
`;

exports[`Download: reducer clear URL 1`] = `
Object {
  "error": null,
  "isOpen": false,
  "url": null,
}
`;

exports[`Download: reducer create user on failure 1`] = `
Object {
  "error": "ERROR",
  "isOpen": false,
  "url": null,
}
`;

exports[`Download: reducer create user on success 1`] = `
Object {
  "error": null,
  "isOpen": false,
  "url": "https://strato-external-storage.s3.amazonaws.com/1529915329415-widescreen.jpeg",
}
`;

exports[`Download: reducer reset Error 1`] = `
Object {
  "error": null,
  "isOpen": false,
  "url": null,
}
`;

exports[`Download: reducer set initial state 1`] = `
Object {
  "error": null,
  "isOpen": false,
  "url": null,
}
`;
